### PR TITLE
Convert Miscellaneous Enzo physics-subdirectories into subcomponents

### PR DESCRIPTION
**Pull request by mabruzzo**
### Pull request summary

This refactors the `Enzo/chemistry` and `Enzo/fluid-props` into distinct components. This is part of a general restructuring that I'm applying to the whole enzo-layer (The changes are directly analogous to those in PRs #332 - #336 )

### Describe your pull request

In more detail, the ultimate goal is to help improve recompilation times.
- At the moment, this PR probably won't help by itself (a lot of other components depend on these)
- However, after I restructure `Enzo/enzo-core` and `Enzo/cosmology` (the last 2 remaining subdirectories), the work done in this PR should definitely improve recompilation times.

**EDIT:** This PR previously tried to restructure `Enzo/cosmology`, but the necessary changes to accomplish that are a little more invasive

### CODE REVIEW

**Comment by mabruzzo** [comment]
I understand the problem. It's a simple fix

**Review by jwise77 — APPROVED** [review]

File: src/Enzo/enzo-core/EnzoConfig.hpp [hunk 1]
```diff
@@ -12,6 +12,10 @@
 #define MAX_FIELDS      30
 #define MAX_FILE_GROUPS 10
 
+// we need to include this for the declaration of GrackleChemistryData
+// -> after GitHub PR #342 is merged, we can delete this line
+#include "Enzo/chemistry/GrackleChemistryData.hpp"
```
Comment (bwoshea) [inline src/Enzo/enzo-core/EnzoConfig.hpp:17]: I think this line can now be safely deleted, correct?

**Review by bwoshea — APPROVED** [review]
I have one small comment - I think something can be deleted - but otherwise it looks good to me.  I'll approve, but @mabruzzo should take a quick look at the line that can be deleted before we merge.

File: src/Enzo/enzo-core/EnzoConfig.hpp (hunk 1 above)
Comment (mabruzzo) [reply src/Enzo/enzo-core/EnzoConfig.hpp]: Yup! You're absolutely right! Thanks for catching this! I removed it. After the tests pass, we can merge

**Pull request merged**